ci-operator/step-registry/upi/install/powervs/cluster/upi-install-powervs-cluster-commands.sh

```diff
@@ -0,0 +1,408 @@
+  do
+    echo "Targetting power cloud instance"
+    retry "ic pi st ${CRN}"
+
+    echo "Deleting the PVM Instances"
+    for INSTANCE_ID in $(ic pi ins --json | jq -r '.pvmInstances[].pvmInstanceID')
+    do
+      echo "Deleting PVM Instance ${INSTANCE_ID}"
+      retry "ic pi ind ${INSTANCE_ID} --delete-data-volumes"
+      sleep 60
+    done
+
+    echo "Deleting the Images"
+    for IMAGE_ID in $(ic pi imgs --json | jq -r '.images[].imageID')
+    do
+      echo "Deleting Images ${IMAGE_ID}"
+      retry "ic pi image-delete ${IMAGE_ID}"
+      sleep 60
+    done
+
+    echo "Deleting the Network"
+    for NETWORK_ID in $(ic pi nets --json | jq -r '.networks[].networkID')
+    do
+      echo "Deleting network ${NETWORK_ID}"
+      retry "ic pi network-delete ${NETWORK_ID}"
+      sleep 60
+    done
+
+    retry "ic resource service-instance-update ${CRN} --allow-cleanup true"
+    sleep 30
+    retry "ic resource service-instance-delete ${CRN} --force --recursive"
+    for COUNT in $(seq 0 5)
+    do
+      FIND=$(ic pi sl 2> /dev/null| grep "${CRN}" || true)
+      echo "FIND: ${FIND}"
+      if [ -z "${FIND}" ]
+      then
+        echo "service-instance is deprovisioned"
+        break
+      fi
+      echo "waiting on service instance to deprovision ${COUNT}"
+      sleep 60
+    done
+    echo "Done Deleting the ${CRN}"
+  done
+
+  echo "Done cleaning up prior runs"
+}
+
+function create_powervs_service_instance() {
+  local version="${1}"
+  local workspace_name="${2}"
+  local powervs_zone="${3}"
+  local resource_group="${4}"
+
+  SERVICE_NAME=power-iaas
+  SERVICE_PLAN_NAME=power-virtual-server-group
+
+  # create workspace for powervs from cli
+  ic resource service-instance-create "${workspace_name}" "${SERVICE_NAME}" "${SERVICE_PLAN_NAME}" "${powervs_zone}" -g "${resource_group}" --allow-cleanup 2>&1 \
+  | tee /tmp/instance.id
+
+  # Process the CRN into a variable
+  CRN=$(cat /tmp/instance.id | grep crn | awk '{print $NF}')
+  export CRN
+  echo "${CRN}" > "${SHARED_DIR}"/POWERVS_SERVICE_CRN
+  POWERVS_SERVICE_INSTANCE_ID=$(echo "${CRN}" | sed 's|:| |g' | awk '{print $NF}')
+  export POWERVS_SERVICE_INSTANCE_ID
+  echo "${POWERVS_SERVICE_INSTANCE_ID}" > "${SHARED_DIR}"/POWERVS_SERVICE_INSTANCE_ID
+  sleep 30
+
+  # Tag the resource for easier deletion
+  ic resource tag-attach --tag-names mac-power-worker-"${CLEAN_VERSION}" --resource-id "${CRN}" --tag-type user
+
+  # Waits for the created instance to become active... after 10 minutes it fails and exists
+  # Example content for TEMP_STATE
+  # active
+  # crn:v1:bluemix:public:power-iaas:osa21:a/3c24cb272ca44aa1ac9f6e9490ac5ecd:6632ebfa-ae9e-4b6c-97cd-c4b28e981c46::
+  COUNTER=0
+  SERVICE_STATE=""
+  while [ -z "${SERVICE_STATE}" ]
+  do
+    COUNTER=$((COUNTER+1)) 
+    TEMP_STATE="$(ic resource service-instances -g "${resource_group}" --output json --type service_instance  | jq -r '.[] | select(.crn == "'"${CRN}"'") | .state')"
```

Comment:
```suggestion
    TEMP_STATE="$(ic resource service-instance -g "${RESOURCE_GROUP}" "${CRN}" --output json --type service_instance  | jq -r '.state')"
```

Please make this change. The service-instances api is paged.

Reply:
updated pattern